### ISSUE TITLE
Create the functional design of the page, how it will be used

### ISSUE THREAD

**Issue by Kreijstal** [Kreijstal/diytest#1]
In this issue we discuss about how the page functionality should be like.

-  First we need the design of the page, we need to describe how will it look like. And what functionality it will have.

As I said in https://github.com/open-source-ideas/open-source-ideas/issues/182 we want a place where students/users can submit their own exams, and everyone else can test themselves with the exams people submit. 

This is my idea: Users can submit "questions" about a topic. And other users can tag the questions to make them clear what they are about. For this let me provide an example. Questions can be boolean, multiple choice, fill the blanks, and they can contain images, audios, etc.

*Example*
Q: The orbits of the planets around the sun are circular.
[true] [false]

When another user sees this question then they can click on either of the buttons, when they click false, they might also get an explanation on why their answer was wrong/right:

"Wrong: The orbits of the planets are close to circular, but are actually elliptical."

People can comment on the question if they want to discuss something, they can post about trivia or do memes, just like youtube comments.

Users can tag questions so this question will have tags like
[astronomy] ,[basic-astronomy], [boolean] (it means it is a true/false question), [5th grade] (I think?)

People can then just look all questions that have tags [basic astronomy], and they will get a lot of questions about this topic.

Suggestions welcome!

**Comment by Kreijstal** [comment]
So I consider each question should be thought as one entry.
The entry content should be treated as a wiki, basically anybody can edit them but you need to get reputation first, (if you are a new user and have a valuable contribution to make you will still be allowed to comment). This will prevent that basically anyone can mass edit all what they want, this also means that we have to keep a history of the entries. Like stackoverflow, basically.

**Comment by keerthivasan-r** [comment]
Sounds like we are having a bigger project. I get this idea. so, basically users can create their own exams for these set of questions? Any tools that we can use to create design mock ups?

**Comment by FredrikAugust** [comment]
Design mock-ups? [Figma](https://www.figma.com/)! @keerthivasan-r

**Comment by Kreijstal** [comment]
@keerthivasan-r basically, users can input questions
other users can take the entry id of the question to make their own exam, or you can browse questions to no end.

If I had to make a booru analogy, a question is an image, and an exam is an album, people can create their own exams, their own album of questions, even if they are not the authors of the questions.

**Comment by Kreijstal** [comment]
I think the input of the questions can be something like this https://github.com/open-source-ideas/open-source-ideas/issues/181 but with a section that explains the reasoning of the questions, I think another issue can be created for this

**Comment by keerthivasan-r** [comment]
I just read the other idea. so, we will have to create a markdown parser to generate questions on the fly. 

Questions
1) Does the user need to know markdown to create a question in the website? if yes, we will have to give mappings from the markdown semantics to our parser semantics?
2) Don't we support creating question just with UI?

@Kreijstal @FredrikAugust your thoughts?

**Comment by keerthivasan-r** [comment]
> Design mock-ups? [Figma](https://www.figma.com/)! @keerthivasan-r

It looks good. I'm not a designer though. Who is going to help for the UI?

**Comment by Kreijstal** [comment]
> I just read the other idea. so, we will have to create a markdown parser to generate questions on the fly.
> 
> Questions
> 
>     1. Does the user need to know markdown to create a question in the website? if yes, we will have to give mappings from the markdown semantics to our parser semantics?
> 
>     2. Don't we support creating question just with UI?
> 
>
> @Kreijstal @FredrikAugust your thoughts?

I am all in for user usability, but I believe that allowing users to learn markdown will be a bit harder for users to learn, but as long as coding is concerned it will be easier to allow markdown input as making a complete user interface where users create the questions. 

This will avoid creating a user interface which means it is easier to mantain, which means less bugs. And it will be faster to code.

**Comment by keerthivasan-r** [comment]
yes, right! I am able to understand. As long as we are going to target students and teachers, I personally prefer to go with UI for question creation. we can also have other ways of creating bulk question sets like markdown, json etc

**Comment by keerthivasan-r** [comment]
One more thing, Are we planning to use something like `jekyll`  (static site generator) by inputting markdown files? For the functionality to make, i think going out for dynamic site is good choice.

What do you guys think?

**Comment by FredrikAugust** [comment]
@Kreijstal, I sadly won't be able to participate in the development of your product. However, if you need anything organized, or want any other administrative help, let me know.

**Comment by Kreijstal** [comment]
> One more thing, Are we planning to use something like `jekyll` (static site generator) by inputting markdown files? For the functionality to make, i think going out for dynamic site is good choice.
> 
> What do you guys think?

Well, we could either do a static site with an API backend, or a dynamic site. Whatever is easiest, and fastest. We probably should work on an API first before worrying about how the website will look like, however to create the API we need to know how users intend to use the website.

**Comment by keerthivasan-r** [comment]
That's right! I would prefer to go with SPA and API back end architecture. Will you be able to create a set of API that the website need to support?

**Comment by Kreijstal** [comment]
> That's right! I would prefer to go with SPA and API back end architecture. Will you be able to create a set of API that the website need to support?

That is the plan, but we first need to have an idea of how the website is supposed to be used.

**Comment by ariadev79** [comment]
Are we drawing a basic sketch of the UI design to see how the user interact?

**Comment by souravghosh97** [comment]
intereseted. Is the project still going on

**Comment by keerthivasan-r** [comment]
Thanks @souravghosh97 @WeiXinFam we need more people for this project to take off. First a UI prototype should be the place to start with

**Comment by VAJRESH** [comment]
I am interested. Is this project still on?

**Comment by Kreijstal** [comment]
> I am interested. Is this project still on?

I mean, it is a lot of work, so that puts a lot of people away, but sure, do you have anything to show?

**Comment by VAJRESH** [comment]
Lets connect and discuss the basic flow of the app, create some basic design and lets get started on development. Things will become clear as we go ahead.
